File: 0412-fizz-buzz/0412-fizz-buzz.java
class Solution {
    public List<String> fizzBuzz(int n) {
        // Create a return list, initialize to number of items
        List<String> answer = new ArrayList<>(n);

        for (int i = 1; i <= n; i++){
            // Check if divisible by 3 or 5
            boolean divisibleBy3 = i % 3 == 0;
            boolean divisibleBy5 = i % 5 == 0;

            if (divisibleBy3 && divisibleBy5) {
                answer.add("FizzBuzz");
            } else if (divisibleBy3) {
                answer.add("Fizz");
            } else if (divisibleBy5) {
                answer.add("Buzz");
            } else {
                answer.add(String.valueOf(i));
            }
        }
        return answer;
    }

    // Time Complexity: O(n)
    // Space Complexity: O(1)
}

File: 0412-fizz-buzz/0412-fizz-buzz.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
    let answer = [];

    for(let i = 1; i <= n; i++){
        let currStr = "";

        if(i % 3 == 0 && i % 5 == 0){
            currStr += "FizzBuzz";
        } else if (i % 3 == 0) {
            currStr += "Fizz";
        } else if (i % 5 == 0){
            currStr += "Buzz";
        } else {
            currStr += i
        }

        answer.push(currStr);

    }
    return answer;
};